core/src/main/scala/slog/StructureEncoder.scala

```diff
@@ -0,0 +1,46 @@
+package slog
+
+trait StructureEncoder[T] {
+  def encode[O](value: T)(implicit structureBuilder: StructureBuilder[O]): O
```

Comment:
i would use context obunds here , no need to give it a name ;)

Reply:
not important right now